SymbolicLinkPreservingUntarTransform failed with FileAlreadyExistsException

The following build failed on CI: 
https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+7.x+bwc/BWC_VERSION=6.8.5,nodes=centos-7&&immutable/301/console

with the stacktrace:

```
Caused by: org.gradle.api.UncheckedIOException: java.nio.file.FileAlreadyExistsException: /var/lib/jenkins/.gradle/caches/transforms-2/files-2.1/3c43923c5dc54e03f90bb76d7dddc3f2/elasticsearch-6.8.5.tar.gz/.
	at org.gradle.internal.UncheckedException.throwAsUncheckedException(UncheckedException.java:61)
	at org.gradle.internal.UncheckedException.throwAsUncheckedException(UncheckedException.java:41)
	at org.elasticsearch.gradle.transform.UnpackTransform.transform(UnpackTransform.java:61)

```
Build scans: 
https://gradle-enterprise.elastic.co/s/sd4qtg3zecjze
https://gradle-enterprise.elastic.co/s/qafb2d6gejmcy

I'm opening this issue in case `SymbolicLinkPreservingUntarTransform` needs some adjustments.

--- a/buildSrc/src/main/java/org/elasticsearch/gradle/transform/SymbolicLinkPreservingUntarTransform.java
+++ b/buildSrc/src/main/java/org/elasticsearch/gradle/transform/SymbolicLinkPreservingUntarTransform.java
@@ -37,6 +37,8 @@ import static org.elasticsearch.gradle.util.PermissionUtils.chmod;
 
 public abstract class SymbolicLinkPreservingUntarTransform implements UnpackTransform {
 
+    private static final Path CURRENT_DIR_PATH = Paths.get(".");
+
     public void unpack(File tarFile, File targetDir) throws IOException {
         Logging.getLogger(SymbolicLinkPreservingUntarTransform.class)
             .info("Unpacking " + tarFile.getName() + " using " + SymbolicLinkPreservingUntarTransform.class.getSimpleName() + ".");
@@ -47,7 +49,7 @@ public abstract class SymbolicLinkPreservingUntarTransform implements UnpackTran
         TarArchiveEntry entry = tar.getNextTarEntry();
         while (entry != null) {
             final Path relativePath = pathModifier.apply(entry.getName());
-            if (relativePath == null) {
+            if (relativePath == null || relativePath.getFileName().equals(CURRENT_DIR_PATH)) {
                 entry = tar.getNextTarEntry();
                 continue;
             }

